inginious/frontend/lti_grade_manager.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+""" Implements LTI AGS. """
+from datetime import datetime
+import logging
+import threading
+import queue
+import time
+
+from pymongo import ReturnDocument
+
+from pylti1p3.contrib.flask import FlaskMessageLaunch
+from pylti1p3.grade import Grade
+from pylti1p3.lineitem import LineItem
+from pylti1p3.launch_data_storage.base import LaunchDataStorage
+
+
+class MongoLTILaunchDataStorage(LaunchDataStorage):
```

Comment:
This could be used to store only the LTI Launch ID in the user session and retrieve the rest from the database.